### ISSUE TITLE
allowAccess: validate uids

### ISSUE THREAD

**Issue by trombonekenny** [PrairieLearn/PrairieLearn#1334]
From Lawrence (but he wasn't the only instructor to do this in the last week):

> I had put [ "netid" ] rather than ["netid@illinois.edu"]
It would be nice to see a warning e.g. on  the assessment list that UID is not enrolled in the course /  not known in the system / not a valid adi.

Should we do validation on sync that uid's exist in the database-- maybe not a fail, but a warning? Or just more reporting when viewing the instructorAssessmentAccess page? Or just write it off as user error and not try to intervene?

**Comment by mwest1066** [comment]
My suggestion is:
1. On sync, if we are in dev mode then do nothing (we don't have a live DB to check validity). If we are in production mode then print a warning for unknown UIDs.
2. On the access control page, color unknown UIDs in red with a pop-up question mark button next to them to explain the problem, or something like that.

**Comment by trombonekenny** [comment]
Yeah. It might be perfectly normal to have uids in the assessment that don't exist on the system yet. I don't really want to put more in the sync screen that just gets ignored.

But the access control page flagging not-ready uids makes perfect sense to me.

**Comment by mwest1066** [comment]
See #3342 for another approach to fix this

**Comment by nwalters512** [comment]
Modern access control doesn't store UIDs in JSON; instructors directly select enrolled students.
